### ISSUE TITLE
Announce the new features: Detailed Action Indicator, Highlighting and Cycling of the Target at Point

### ISSUE THREAD

**Issue by minad** [oantolin/embark#291]
![gif](https://user-images.githubusercontent.com/50754/127231275-5193fb62-0e62-4e69-9847-a35d0b8e9ff2.png)

![movie](https://user-images.githubusercontent.com/50754/127237854-a95ade6e-f29e-4d68-abf4-887397126fd3.gif)

**Comment by oantolin** [comment]
Yes, an announcement is desperately needed. Definitely some changes to the README, maybe also a reddit post?

**Comment by oantolin** [comment]
Also a new tag release (although I want to wait a couple of days for that in case of undiscovered bugs).

**Comment by minad** [comment]
Okay, do you mind if I make a post with the recent news? I think it would be good to have a post now while the recent developments are still new. And then after you tagged a new release another announcement post listing more changes in detail. I am kind of excited by the recent changes and I am eager to hear what people think.

**Comment by minad** [comment]
If you want to post it only once that's okay - some projects post a lot of updates, some only releases. I am just unreasonably excited by the new developments :sweat_smile:

**Comment by oantolin** [comment]
Go ahead! To post one reddit I don't think we need to wait.

**Comment by oantolin** [comment]
These three things together really add a whole new level of polish to Embark. It feels more professional now, like Consult. 😉

**Comment by minad** [comment]
Posted here: https://www.reddit.com/r/emacs/comments/osx5t9/recent_embark_developments_detailed_action/

Well Consult *is* professional, I mean already the name. It is supposed for enterprise Emacs installs or professional academic installs :smile:

**Comment by minad** [comment]
Hopefully we get some interesting feedback :)

**Comment by oantolin** [comment]
What is going on at the top now? I see two copies of the gif.

**Comment by minad** [comment]
I abused github as a gif filehoster :-P

**Comment by jdtsmith** [comment]
These are great!  So glad my suggestion of highlighting the match made it in.  This really adds to the functionality/discoverability.  Now if we can pre-highlight forward/backward showing matches (like isearch does) I'll never need `M-s . C-s` again.  Perhaps `n` / `p` can enable _temporary_ highlighting of the symbol at point?

**Comment by oantolin** [comment]
I'm not sure how much special treatment the next/previous symbol actions should get...

In the meantime, if you want to highlight occurrences of the symbol at point there is a toggle highlight action bound to capital `H`.

**Comment by jdtsmith** [comment]
I saw `H` which looks useful, but it leaves the highlighting active and then exits, whereas I'd usually be after the opposite: temporarily highlight and then immediate _nearby_ navigation with `n`/`p`.  (I _love_ that you can then convert this into a `consult-line`!).   Now that embark has non-exiting actions available from outside the minibuffer (what the `*` means?), perhaps it makes sense to allow non-exiting actions with prefix.  E.g. `C-u H` to turn on hi-lighting but _not_ exit.  That would get me part the way there.

**Comment by oantolin** [comment]
For now, you can add `embark-toggle-highlight` to `embark-repeat-commands` if you never want it to exit `embark-act`. If you want to have the choice of exiting or not, you can define an alias and make that one repeatable

```elisp
(defalias 'embark-toggle-highlight-noexit 'embark-toggle-highlight)
(add-to-list 'embark-repeat-commands 'embark-toggle-highlight-noexit)
(define-key embark-identifier-map (kbd "M-h") 'embark-toggle-highlight-noexit)
```

I realize this is not as comfortable as having the highlighting automatic, since you need to turn it on and off.

As for having the prefix indicate non-exiting, as in your `C-u H` example, I think that's probably not a good idea. Currently the prefix argument means, pretty logically, to run the action with a prefix argument, and some actions definitely need the option of using their prefix argument. Maybe the thing to do would be to have a separate key function as a kind of prefix argument for this, maybe `M-c` for "continue", so `M-c` before any action would mark it as temporarily repeatable.

**Comment by oantolin** [comment]
Another thing you could do, that's more automatic is to have the symbol target finder highlight all occurrences of the the symbol at point and unhighlight them in a post-command hook:

```elisp
(defun temporarily-highlight-symbol-at-point (target)
  (when target
    (let ((regexp (find-tag-default-as-symbol-regexp))
          (hook (make-symbol "hook")))
      (fset 'hook (lambda ()
                    (remove-hook 'post-command-hook 'hook)
                    (unhighlight-regexp regexp)))
      (add-hook 'post-command-hook hook)
      (highlight-symbol-at-point))
    target))

(advice-add 'embark-target-identifier-at-point
            :filter-return 'temporarily-highlight-symbol-at-point)
```

This is pretty close to what you want for just symbol targets. As far as I can tell the only problem is that the highlighting sticks around even if you cycle to other targets.

**Comment by minad** [comment]
We should not add special treatment for symbols here. This would go very much against the Zen of Embark as I see it ;)

But we could consider adding general hooks target-activated-hook? It could even be an alist with the type as key. In principle this is very much the same as the indicator. What about making embark-indicator an alist with type as key and hooks as value?

**Comment by minad** [comment]
I really like this idea btw, right now embark--highlight target is ad-hoc functionality, it should instead also be implemented as a general action indicator, such that people can turn off this annoying highlighting if they want ;)

**Comment by jdtsmith** [comment]
> Currently the prefix argument means, pretty logically, to run the action with a prefix argument, and some actions definitely need the option of using their prefix argument. 

I wondered about that.  In fact I had a related issue regarding embark and prefixes.  It doesn't seem prefixes to export get passed on to the exported command.  For example I regularly use (with isearch/occur) `M-s .`, `M-5 M-s o` to get 5 lines of context in an occur buffer.  When exporting from consult-line etc. to an occur buffer, I had hoped a `M-5 C-, E` would accomplish the same thing, but the prefix info doesn't seem to make it through. 

Thanks both for your work on this (which I mostly missed while messing around with consult-imenu!).

**Comment by jdtsmith** [comment]
I'll open a separate issue.
